### DISCUSSION TITLE
Scan destination safety (like in sqlx)

### DISCUSSION THREAD

**Discussion by josephbuchma** [go-jet/jet#369] [Q&A]
Is something like [this](https://jmoiron.github.io/sqlx/#safety) possible in jet?

> By default, StructScan will return an error if a column does not map to a field in the destination.

**Comment by go-jet** [comment]
No, there can be more destinations then columns, and there can be more columns then destinations.

**Reply by josephbuchma** [reply]
What do you think about this feature in the context of jet? It's less likely to make a mapping mistake with jet's query builder & generated models, but definitely possible. I've ween bitten by this few times already.

For example, one thing I keep forgetting is that when you embed a struct, you still have to use type name of the embedded struct. Would love to get an error when this happens :)

**Reply by go-jet** [reply]
Yeah, I agree such a feature would be beneficial. The only concern is will it potentially brake mapping for existing users with more columns then destinations queries. The solution might be to add a new strict query method, or some config for strict mapping, or just make a change and hope that users with bad queries have test coverage over all their queries and CI setup.
